excel/src/main/java/org/apache/metamodel/excel/ExcelUtils.java

```diff
@@ -281,6 +356,51 @@ private static String getFormulaCellValue(Workbook wb, Cell cell) {
         return cell.getCellFormula();
     }
 
+    private static Object getFormulaCellValueAsObject(final Workbook workbook, final Cell cell) {
+        // first try with a cached/precalculated value
+        try {
+            return getDoubleAsNumber(cell.getNumericCellValue());
+        } catch (final Exception e) {
+            if (logger.isInfoEnabled()) {
+                logger.info("Failed to fetch cached/precalculated formula value of cell: " + cell, e);
+            }
+        }
+
+        // evaluate cell first, if possible
+        try {
```

Comment:
This piece of code looks like a carbon copy of line 326-348, with the only difference that in those lines, on line 339 you invoke `getCellValue`, where here on line 377 you invoke `getCallValueAsObject`,  I would propose you refactor this to get rid of the duplicated boiler plate code around these invocations (so it's no longer duplicated, just available once).

Reply:
ok, I've put it into a method